File: eval.py
#-----------------------------------------------------------------------------
# eval helper (Python)
#-----------------------------------------------------------------------------
import sys
import re
import codecs
import StringIO

# Host-Script interface
# utils

_result = ""
_n = 0

__stdBuffer = None


#######################################################################
# eval helper

def ASConsole_execEval(_text):
	global _result, _n
	_r = None
	try:
		if _text[0:1] == ":":
			_r = eval(_text[1:], globals())
		else:
			exec(_text, globals())
			return None
	except:
		utils.setError(repr(sys.exc_info()))
		return None
	if utils.getError() is not None:
		return None
	_result = _r
	if isinstance(_r, (int, long, float)):
		_n = _r
	if isinstance(_r, unicode):
		return _r
	elif isinstance(_r, str):
		return _r.decode('utf-8')
	else:
		return repr(_r)


#######################################################################

def exportVariable(name, value):
	utils.setVariable(name, value)

def importVariable(name):
	return utils.getVariable(name)


#######################################################################

def comInfo(obj, member=None):
	utils.comInfo(obj, member)

def comMethods(obj):
	utils.comMethods(obj)

def comProperties(obj):
	utils.comProperties(obj)

def comProps(obj):
	utils.comProperties(obj)


#######################################################################
# internal functions (interface)

def ASConsole_onActive():
	pass

def ASConsole_setLocale():
	pass

def ASConsole_isEnableStdIO():
	return True

def ASConsole_initStdBuffer():
	global __stdBuffer
	__stdBuffer = StringIO.StringIO()
	sys.stdout = __stdBuffer
	sys.stderr = __stdBuffer

def ASConsole_getStdBuffer():
	global __stdBuffer
	try:
		buf = __stdBuffer
		ASConsole_initStdBuffer()
		result = buf.getvalue()
		buf.close()
		return __getUnicodeText(result)
	except:
		utils.setError(repr(sys.exc_info()))
		return None


#######################################################################
# internal functions

# temporary step
def __getUnicodeText(s):
	try: return s.decode('utf-8')
	except: pass
	try: return s.decode('shift_jis')
	except: pass
	return s


#######################################################################

ASConsole_initStdBuffer()

File: eval.php
#-----------------------------------------------------------------------------
# eval helper (PHPScript)
#-----------------------------------------------------------------------------

# Host-Script interface
# $utils

$GLOBALS["user_variables"] = array();


#######################################################################
# eval helper

function ASConsole_execEval($__text)
{
	# special convert
	if (ereg('^[0-9]', $__text) || ereg('^[$][A-Za-z_][A-Za-z0-9_]*$', $__text)) {
		$__text = "print($__text);";
	}
	else {
		$__text = ereg_replace('^p +([^(].*)$', 'p(\1);', $__text);
	}

	extract($GLOBALS["user_variables"]);

	$__result = eval($__text);

	$__vars = get_defined_vars();
	$__keys = array_keys($__vars);
	for ($__i = 0; $__i < count($__keys); $__i++) {
		if (ereg('^__', $__keys[$__i])) {
			unset($__vars[$__keys[$__i]]);
		}
	}
	$GLOBALS["user_variables"] = $__vars;

	return $__result;
}

#######################################################################

function exportVariable($name, $value)
{
	global $utils;
	$utils->setVariable($name, $value);
}

function importVariable($name)
{
	global $utils;
	return $utils->getVariable($name);
}


#######################################################################

function p()
{
	for($i = 0; $i < func_num_args(); $i++) {
		var_dump(func_get_arg($i));
	}
}


#######################################################################

function comInfo($obj, $member=NULL)
{
	global $utils;
	$utils->comInfo($obj, $member);
}

function comMethods($obj)
{
	global $utils;
	$utils->comMethods($obj);
}

function comProperties($obj)
{
	global $utils;
	$utils->comProperties($obj);
}

function comProps($obj)
{
	global $utils;
	$utils->comProperties($obj);
}


#######################################################################
# internal functions (interface)

function ASConsole_onActive()
{
}

function ASConsole_setLocale()
{
}

function ASConsole_isEnableStdIO()
{
	return true;
}

function ASConsole_initStdBuffer()
{
	ob_clean();
}

function ASConsole_getStdBuffer()
{
	$buf = ob_get_contents();
	ASConsole_initStdBuffer();
	return $buf;
}


#######################################################################
# internal functions

function __error_handler($err_code, $err_string, $filename, $line, $symbols)
{
	$utils->setError($utils->createError($err_code, $err_string));
}


#######################################################################

#error_reporting(E_ALL & ~E_NOTICE);
#set_error_handler('__error_handler');

ob_start();
